sklearn/metrics/_pairwise_distances_reduction/_middle_term_computer.pyx.tp

```diff
@@ -27,35 +25,82 @@ from ...utils._cython_blas cimport (
   Trans,
   _gemm,
 )
+from ...utils._typedefs cimport DTYPE_t, ITYPE_t
+
+from scipy.sparse import issparse
+
 
 {{for name_suffix, upcast_to_float64, INPUT_DTYPE_t, INPUT_DTYPE in implementation_specific_values}}
 
-cdef class GEMMTermComputer{{name_suffix}}:
-    """Component for `EuclideanDistance` specialisation wrapping the logic for the call to GEMM.
 
-    `EuclideanDistance` subclasses internally compute the squared Euclidean distances
-    between chunks of vectors X_c and Y_c using the following decomposition:
+cdef class MiddleTermComputer{{name_suffix}}:
+    """Abstract base class for the `EuclideanDistance` specialisation
+    components wrapping the computation of the distance matrix chunks.
```

Comment:
PEP 257 style: first line of multiline docstring should be a one-liner followed by a blank line.

```suggestion
    """Helper class to compute a Euclidean distance matrix in chunks . 
    
    This is an abstract base class that is further specialized depending
    on the type of data (dense or sparse).
```